pkg/controller/controller.go

```diff
@@ -268,27 +284,27 @@ func NewCSIProvisioner(client kubernetes.Interface,
 	return provisioner
 }
 
-// This function get called before any attepmt to communicate with the driver.
+// This function get called before any attempt to communicate with the driver.
 // Before initiating Create/Delete API calls provisioner checks if Capabilities:
 // PluginControllerService,  ControllerCreateVolume sre supported and gets the  driver name.
-func checkDriverState(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (string, error) {
-	ok, err := supportsPluginControllerService(grpcClient, timeout)
+func checkDriverState(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (*driverState, error) {
+	pluginCaps, err := getPluginCapabilities(grpcClient, timeout)
 	if err != nil {
-		glog.Errorf("failed to get support info :%v", err)
-		return "", err
+		glog.Errorf("failed to get plugin capabilities: %v", err)
```

Comment:
This should be handled as a separate pr, but I want to get rid of this style of logging errors that we're also returning

Reply:
I'll address it in this PR. Improving the code base one line at a time :)